inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -638,7 +638,25 @@ void ReversePermutationsPass::run() {
 
 void RemovePermutationsNHWCToNCHWPass::run() {
     std::list<CNNLayerPtr> permutationsToRemove;
-
+    auto isPermuteAroundConvolution = [](CNNLayerPtr layer, bool before) {
+        auto nameEndsWith = [layer](const std::string & suffix) {
+            return layer->name.size() >= suffix.size() &&
+                layer->name.compare(layer->name.size() - suffix.size(), suffix.size(), suffix) == 0;
+        };
+        if (before && !nameEndsWith("/input_transpose") || !before && !nameEndsWith("/output_transpose"))
```

Comment:
Model optimizer can insert permutes not only with --disable_nhwc_to_nchw option but without it for any framework, for example before and after reshapes. If there are reshapes before and after convolution there are no other differences from GNA plugin point of view except naming. I agree that it's not a good way to check a name but while I don't see another way to understand that we should remove these permutes and their removing causes errors for models created in some frameworks.

Reply:
Actually model optimizer can insert permutes without --disable_nhwc_to_nchw option but only when it converts from NHWC format. But the original model can contain a reshape from 1, 160 (NC) -> 1, 1, 1, 160 (NCHW) and it is not a permute. Another original model can contain reshape 1, 160 (NC) -> 1, 1, 160, 1 (NHWC) and MO will add a permute 1, 1, 160, 1 (NHWC) -> 1, 1, 1, 160 (NCHW) that will be fused to one reshape 1, 160 (NC) -> 1, 1, 1, 160 (NCHW). And there is no a way to distinguish what case we have.

Reply:
I find hardcoding names unacceptable, could you give me name of some model that will fail in case that you describe? If it is NCHW and is trivial it should not matter if we change it NHWC and treat it as such since the data in memory is laid exactly the same, especially if the layer after is permute (we could check if the order of permutes match though, since we know what is expected ordering of inputs and outputs).